src/main/java/com/google/api/generator/gapic/composer/ServiceClientSampleCodeComposer.java

```diff
@@ -169,4 +179,169 @@ public static String composeClassHeaderEndpointSampleCode(
             ExprStatement.withExpr(initSettingsVarExpr),
             ExprStatement.withExpr(initClientVarExpr)));
   }
+
+  public static String composeRpcMethodHeaderSampleCode(
+      Method method,
+      List<MethodArgument> arguments,
+      TypeNode clientType,
+      Map<String, ResourceName> resourceNames) {
+    // TODO(summerji): Add Other types RPC methods' sample code.
+    return SampleCodeWriter.write(
+        composeUnaryRpcMethodSampleCode(method, arguments, clientType, resourceNames));
+  }
+
+  public static TryCatchStatement composeUnaryRpcMethodSampleCode(
+      Method method,
+      List<MethodArgument> arguments,
+      TypeNode clientType,
+      Map<String, ResourceName> resourceNames) {
+    VariableExpr clientVarExpr =
+        VariableExpr.withVariable(
+            Variable.builder()
+                .setName(JavaStyle.toLowerCamelCase(clientType.reference().name()))
+                .setType(clientType)
+                .build());
+    // Assign each method arguments with its default value.
+    List<Expr> methodArgExprs = createMethodArgVarExprs(arguments, resourceNames);
```

Comment:
Here's another approach to doing this. We replace the index-based coupling between the list of MethodArguments and VariableExprs, and instead use a coupling that makes more semantic sense - those of the AssignmentExprs. We’ll also streamline the ordering, and rely less on circumstantial checks. Let's put aside the method wrapping and start by focusing on the core logic that needs to happen.

```java
List<Expr> rpcMethodArgVarExprs = arguments.stream()
        .map(arg -> 
             VariableExpr.withVariable(Variable.builder()
                          .setName(JavaStyle.toLowerCamelCase(arg.name()))
                          .setType(arg.type())
                          .build()))
        .collect(Collectors.toList());
List<Expr> rpcMethodArgValueExprs = arguments.stream()
    .map(arg ->
          isStringTypedResourceName(arg) 
          ? DefaultValueComposer.createDefaultValue(
                 resourceNames.get(arg.field().resourceReference().resourceTypeString()))
          : DefaultValueComposer.createDefaultValue(arg, resourceNames))
       .collect(Collectors.toList());

List<Expr> bodyExprs = new ArrayList<>();
Preconditions.checkState(rpcMethodArgVarExprs.size() == rpcMethodArgValueExprs.size(), “Error message here”);
bodyExprs.addAll(
    IntStream.range(0, rpcMethodArgVarExprs.size())
    .mapToObj(i -> AssignmentExpr.builder().setVariableExpr(rpcMethodArgVarExprs.get(i)).setValueExpr(rpcMethodArgValueExprs.get(i)).build())
    .collect(Collectors.toList()));
    
```

Reply:
Meant to also add this:

```java
private boolean isStringTypedResourceName(MethodArgument arg, 
    Map<String, ResourceName> resourceNames) {
    return arg.type().equals(TypeNode.STRING)
        && arg.field().hasResourceReference()
        && resourceNames.containsKey(
        arg.field().resourceReference().resourceTypeString());
}
```

Reply:
Many thanks on discussion of default value expr for method arguments in the chat.
SG.